projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c

```diff
@@ -0,0 +1,1806 @@
+					    "                    <opt> = PGA option; values are: opt0, opt1, ... opt7 corresponding to the datasheet.\n\r"
+					    "                    Example: aps 0 opt3 - set ADC channel 0 to PGA 3, gain value of 8.\n\r"
+					    " apg <chan>       - Display a channel's PGA option; return values are: opt0, opt1, ... opt7 corresponding to the datasheet.\n\r"
+					    "                    <chan> = ID of the channel to be read.\n\r"
+					    "                    Example: apg 0 - read the PGA value of channel 0.\n\r"
+					    " aos <odr>        - Set ADC sample rate. Filter option, power mode and reference clock must be taken into consideration.\n\r"
+					    "                    <odr> = New sample rate value.\n\r"
+					    "                    Example: aos 2000 - set sample rate to 2000 samples per second.\n\r"
+					    " aog              - Read the current sample rate.\n\r"
+					    " afs <opt> <post> - Set ADC filter option.\n\r"
+					    "                    <opt> = filter option; can be: 'sinc4', 'sinc3', 'fflt4', 'fflt3' and 'postf'.\n\r"
+					    "                    <post> = post-filter option; can be: 'opt0', 'opt1', ... 'opt3'; add only when opt=postf.\n\r"
+					    "                    Example: afs fflt4 - set filter option to fast settling sinc4.\n\r"
+					    " afg              - Read the current filter.\n\r");
+	}
+}
+
+/**
+ * Display help options in the CLI.
+ *
+ * @param [in] dev - The device structure.
+ * @param [in] arg - Not used in this case. It exists to keep the function
+ *                   prototype compatible with the other functions that can be
+ *                   called from the CLI.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad7124_8pmdz_help(struct ad7124_8pmdz_dev *dev, uint8_t *arg)
+{
+	int32_t ret;
+
+	ret = ad7124_8pmdz_help_prompt(dev, HELP_LONG_COMMAND);
+	if(ret != 0)
+		return ret;
+
+	ret = ad7124_8pmdz_help_general(dev, HELP_LONG_COMMAND);
+	if(ret != SUCCESS)
+		return ret;
+
+	ret = ad7124_8pmdz_help_adc(dev, HELP_LONG_COMMAND);
+	if(ret != SUCCESS)
+		return ret;
+
+	ret = ad7124_8pmdz_help_prompt(dev, HELP_SHORT_COMMAND);
+	if(ret != 0)
+		return ret;
+
+	ret = ad7124_8pmdz_help_general(dev, HELP_SHORT_COMMAND);
+	if(ret != SUCCESS)
+		return ret;
+
+	return ad7124_8pmdz_help_adc(dev, HELP_SHORT_COMMAND);
+}
+
+/**
+ * @brief CLI command to read and display a device register.
+ * @param [in] dev - Pointer to the application handler.
+ * @param [in] arg - The address of the register to be displayed.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad7124_8pmdz_reg_read(struct ad7124_8pmdz_dev *dev, uint8_t *arg)
+{
+	uint8_t reg_addr, i;
+	struct ad7124_st_reg *reg_struct_ptr;
+	int32_t ret;
+	uint8_t buff[50];
+
+	reg_addr = strtol((char *)arg, NULL, 16);
+
+	for(i = 0; i < AD7124_REG_NO; i++) {
+		if(dev->ad7124_handler->regs[i].addr == reg_addr) {

```

Comment:
I think you shouldn't access what is inside dev.
A better approach will be to iterate throught an array like this: https://github.com/analogdevicesinc/no-OS/blob/0a574cbbbc092523741b9fd8c7f6ff88d9770597/drivers/adc/ad7124/ad7124_regs.c#L44

Reply:
That type of array needs to be attached to the handler in the setup routine.